aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,861 @@
+    use aptos_framework::resource_account;
+    use aptos_framework::timestamp;
+    use aptos_token::token::{Self, TokenMutabilityConfig, create_token_mutability_config, create_collection, create_tokendata, TokenId, TokenDataId};
+    use aptos_std::bucket_table::{Self, BucketTable, new};
+    use std::bcs;
+
+    struct NFTMintConfig has key {
+        admin: address,
+        treasury: address,
+        signer_cap: SignerCapability,
+        token_minting_events: EventHandle<MintingEvent>,
+        token_exchange_events: EventHandle<ExchangeEvent>,
+    }
+
+    struct CollectionConfig has key {
+        collection_name: String,
+        collection_description: String,
+        collection_maximum: u64,
+        collection_uri: String,
+        collection_mutate_config: vector<bool>,
+        // this is base name, when minting, we will generate the actual token name as `token_name_base: sequence number`
+        token_name_base: String,
+        royalty_payee_address: address,
+        token_description: String,
+        token_maximum: u64,
+        token_uri: vector<String>,
+        token_mutate_config: TokenMutabilityConfig,
+        royalty_points_den: u64,
+        royalty_points_num: u64,
+        property_keys: vector<vector<String>>,
+        property_values: vector<vector<vector<u8>>>,
+        property_types: vector<vector<String>>,
+    }
+
+    struct WhitelistMintConfig has key {
+        whitelisted_address: BucketTable<address, u64>,
+        whitelist_mint_price: u64,
+        whitelist_minting_start_time: u64,
+        whitelist_minting_end_time: u64,
+    }
+
+    struct PublicMintConfig has key {
+        public_mint_price: u64,
+        public_minting_start_time: u64,
+        public_minting_end_time: u64,
+    }
+
+    struct SourceToken has key {
+        token_data_id: TokenDataId,
+    }
+
+    struct MintingEvent has drop, store {
+        token_receiver_address: address,
+        token_id: TokenId,
+    }
+
+    struct ExchangeEvent has drop, store {
+        token_receiver_address: address,
+        token_id: TokenId,
+    }
+
+    const BURNABLE_BY_OWNER: vector<u8> = b"TOKEN_BURNABLE_BY_OWNER";
+
+    const ENOT_AUTHORIZED: u64 = 1;
+    const EINVALID_TIME: u64 = 2;
+    const EACCOUNT_DOES_NOT_EXIST: u64 = 3;
+    const EVECTOR_LENGTH_UNMATCHED: u64 = 4;
+    const EEXCEEDS_COLLECTION_MAXIMUM: u64 = 5;
+    const EINVALID_PRICE: u64 = 6;
+    const EINVALID_UPDATE_AFTER_MINTING: u64 = 7;
+    const EMINTING_IS_NOT_ENABLED: u64 = 8;
+    const ENO_ENOUGH_TOKENS_LEFT: u64 = 9;
+    const EACCOUNT_NOT_WHITELISTED: u64 = 10;
+    const EINVALID_ROYALTY_NUMERATOR_DENOMINATOR: u64 = 11;
+    const ECOLLECTION_ALREADY_CREATED: u64 = 12;
+    const ECONFIG_NOT_INITIALIZED: u64 = 13;
+    const EAMOUNT_EXCEEDS_MINTS_ALLOWED: u64 = 14;
+    const ETOKEN_ID_NOT_FOUND: u64 = 15;
+    const ECANNOT_EXCHANGE_BEFORE_MINTING_ENDS: u64 = 16;
+
+    fun init_module(resource_account: &signer) {
```

Comment:
Why not just name this post_mint_reveal_nft to be clear instead of resource_account?

Reply:
I guess technically this signer is the signer of the resource account though? could be confusing to name it `post_mint_reveal_nft`

Reply:
But doesn't the code here expect them to be the same? That's where the NFTMintConfig resource is stored

Reply:
yeah they're the same, but I think  `resource_account` in `init_module` specifies that the `resource_account` is the module publisher, and the rest of the code refers to `post_mint_reveal_nft` as the module address. so not sure if it'd be confusing to call it the resource signer `post_mint_reveal_nft` here. maybe we can call it `post_mint_reveal_nft_resource_signeer`?

Reply:
It's just confusing in my mind because init_module would be called with the same account that later NFTMintConfig resource will be read from. But they have different names so it feels like 2 different accounts/addresses. Anyway, I don't feel too strongly about this so up to you here :)

Reply:
just updated the name to combine both lol!